src/diffusers/models/unet_2d_condition.py

```diff
@@ -321,6 +322,7 @@ def __init__(
                 attn_num_head_channels=attention_head_dim[-1],
                 resnet_groups=norm_num_groups,
                 resnet_time_scale_shift=resnet_time_scale_shift,
+                only_cross_attention=mid_block_only_cross_attention,
```

Comment:
@patrickvonplaten you had used `only_cross_attention[-1]` for this argument but I think it makes sense to use a separate config for the mid block. The other arguments for the mid block that re-use the config for the last encoder block make sense because they are dimensionality based and they _have_ to match. But this constraint doesn't hold necessarily for the `only_cross_attention` flag

Reply:
Actually with this commit, https://github.com/huggingface/diffusers/pull/3011/commits/7329eadd5dc4accc90763cfe3829ecd3e687dea3 , we can do a better default to the value of `only_cross_attention` when it's given as a single boolean